### PR TITLE
Add autoApply option

### PR DESCRIPTION
**Pull request by patmood**
Add an auto apply option that hides the apply and cancel buttons, and immediately notifies callback when calendar dates change (default is false)

### CODE REVIEW

**Comment by bstst** [comment]
This should be merged in. As this is awesome. Why isn't it merged in?

**Comment by oswaldoacauan** [comment]
:+1:

**Comment by jaketoolson** [comment]
:+1: this would be great!

**Comment by joseph-onsip** [comment]
:+1:

**Comment by rmasters** [comment]
This would be really useful when used with `singleDatePicker`.

File: daterangepicker.js [hunk 1]
```diff
@@ -290,6 +291,11 @@
                 this.timePicker12Hour = options.timePicker12Hour;
             }
 
+            if (typeof options.autoApply === 'boolean') {
+                this.autoApply = options.autoApply;
+                this.container.find('.applyBtn, .cancelBtn').addClass('hide');
```
Comment (nickolas1) [inline daterangepicker.js]: <pre><code>                
if (this.autoApply) {
this.container.find(".applyBtn, .cancelBtn").addClass("hide");
                    if (this.singleDatePicker) {
                        this.container.find(".ranges").addClass("hide");
                    }
                }
</pre></code>

Comment (patmood) [inline daterangepicker.js]: Agreed. I didn't consider the case when autoApply is present but set to false! I'll fix it now.

**Pull request merged**